resources/providers/kubernetes_provider_mock.go

```diff
@@ -0,0 +1,70 @@
+// Code generated by mockery v2.10.4. DO NOT EDIT.
```

Comment:
I didn't realize it uses `codegen` underlying.
We should understand what are the best practices for triggering the code generation in our SDLC.
Meaning, do we want to automate it or keep it a manual process?

Reply:
I added new code guidelines in the Readme.md file